src/modules/dashboard/components/ColonyDecisions/DecisionPreview.tsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import { useParams, Redirect } from 'react-router-dom';
+import { defineMessages } from 'react-intl';
+
+import Heading from '~core/Heading';
+import Button from '~core/Button';
+import Tag from '~core/Tag';
+import HookedUserAvatar from '~users/HookedUserAvatar';
+import {
+  useUser,
+  useLoggedInUser,
+  useColonyFromNameQuery,
+  AnyUser,
+  OneDomain,
+} from '~data/index';
+import { ColonyActions } from '~types/index';
+import { NOT_FOUND_ROUTE } from '~routes/index';
+import LoadingTemplate from '~pages/LoadingTemplate';
+
+import DetailsWidget from '../ActionsPage/DetailsWidget';
+
+import styles from './DecisionPreview.css';
+
+const MSG = defineMessages({
+  preview: {
+    id: 'dashboard.ColonyDecisions.DecisionPreview.preview',
+    defaultMessage: `Preview`,
+  },
+  loadingText: {
+    id: 'dashboard.ColonyDecisions.DecisionPreview.loadingText',
+    defaultMessage: 'Loading Decision',
+  },
+});
+
+/* mock data */
+const decisionData = {
+  title: 'Should we build a Discord Bot?',
+  description: `I think we should build a Discord bot that integrates with the Dapp and provides our community with greater transperency and also provides more convienience for us to be notified of things happening in our Colony.`,
+  actionType: 'Decision',
+  fromDomain: '1',
+};
+
+const handleEdit = () => {};
+const handleSubmit = () => {};
+
+const displayName = 'dashboard.ColonyDecisions.DecisionPreview';
+
+const DecisionPreview = () => {
+  const { colonyName } = useParams<{
+    colonyName: string;
+  }>();
+  const { walletAddress, username } = useLoggedInUser();
+  const userProfile = useUser(walletAddress) as AnyUser;
+  const actionType = decisionData.actionType as ColonyActions;
+
+  const { data: colonyData, error, loading } = useColonyFromNameQuery({
+    // We have to define an empty address here for type safety, will be replaced by the query
+    variables: { name: colonyName, address: '' },
+    pollInterval: 5000,
+  });
+
+  if (
+    loading ||
+    (colonyData?.colonyAddress &&
+      !colonyData.processedColony &&
+      !((colonyData.colonyAddress as any) instanceof Error))
+  ) {
+    return (
+      <div className={styles.loadingWrapper}>
+        <LoadingTemplate loadingText={MSG.loadingText} />
+      </div>
+    );
+  }
+
+  if (!colonyName || error || !colonyData?.processedColony) {
+    console.error('error', error);
+    return <Redirect to={NOT_FOUND_ROUTE} />;
+  }
+
+  if (colonyData?.processedColony) {
```

Comment:
This check is useless as it will never go to this line is there is no processed  colony. It will redirect to the not found route before that. Please, remove.